src/EventStore.Core/GcPauseListener.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Diagnostics.Metrics;
+using System.Diagnostics.Tracing;
+
+namespace EventStore.Core;
+
+public class GcPauseListener : EventListener {
```

Comment:
this is cool, perhaps we can see how long the pauses are by consuming some of the other events too. for simplicity i would probably take it out of this PR though and put it in another one (and it would definitely need to be enabled/disabled separately of the rest)